widget/label.go

```diff
@@ -1,213 +1,23 @@
-		if l.label.color != nil {
-			text.Color = l.label.color
-		} else {
-			text.Color = theme.TextColor()
-		}
-	}
-}
-
-func (l *labelRenderer) BackgroundColor() color.Color {
-	return color.Transparent
-}
-
-func (l *labelRenderer) Refresh() {
-	l.ApplyTheme()
-	for _, text := range l.texts {
-		text.Alignment = l.label.Alignment
-		text.TextStyle = l.label.TextStyle
-	}
-
-	canvas.Refresh(l.label)
-}
-
-// Label widget is a basic text component with appropriate padding and layout.
-type Label struct {
-	baseWidget
-
-	Text      string         // The content of the label
-	Alignment fyne.TextAlign // The alignment of the Text
-	TextStyle fyne.TextStyle // The style of the label text
-
-	color color.Color
-}
-
-// Resize sets a new size for a widget.
-// Note this should not be used if the widget is being managed by a Layout within a Container.
-func (l *Label) Resize(size fyne.Size) {
-	l.resize(size, l)
-}
-
-// Move the widget to a new position, relative to it's parent.
-// Note this should not be used if the widget is being managed by a Layout within a Container.
-func (l *Label) Move(pos fyne.Position) {
-	l.move(pos, l)
-}
-
-// MinSize returns the smallest size this widget can shrink to
-func (l *Label) MinSize() fyne.Size {
-	return l.minSize(l)
-}
-
-// Show this widget, if it was previously hidden
-func (l *Label) Show() {
-	l.show(l)
-}
-
-// Hide this widget, if it was previously visible
-func (l *Label) Hide() {
-	l.hide(l)
-}
-
-// SetText updates the text of the label widget
-func (l *Label) SetText(text string) {
-	l.Text = text
-
-	render := Renderer(l).(*labelRenderer)
-	for _, obj := range render.texts {
-		obj.Text = ""
-	}
-
-	render.updateTexts(render.parseText(l.Text))
-}
-
-// Rows returns the number of text rows in this text entry.
-// The entry may be longer than required to show this amount of content.
-func (l *Label) Rows() int {
-	return Renderer(l).(*labelRenderer).lines
-}
-
-// RowLength returns the number of visible characters in the row specified.
-// The row parameter should be between 0 and l.Rows()-1.
-func (l *Label) RowLength(row int) int {
-	return len(Renderer(l).(*labelRenderer).texts[row].Text)
-}
-
-// CreateRenderer is a private method to Fyne which links this widget to it's renderer
-func (l *Label) CreateRenderer() fyne.WidgetRenderer {
-	render := &labelRenderer{label: l}
-
-	render.texts = []*canvas.Text{}
-	render.objects = []fyne.CanvasObject{}
-	render.updateTexts(render.parseText(l.Text))
-
-	return render
-}
+// Label widget is a label component with appropriate padding and layout.
+type Label = Text
 
 // NewLabel creates a new layout widget with the set text content
 func NewLabel(text string) *Label {
-	var style fyne.TextStyle
+	return NewLabelStyled(text, fyne.TextAlignLeading, fyne.TextStyle{})
+}
 
+// NewLabelStyled creates a new layout widget with the set text content
+func NewLabelStyled(text string, alignment fyne.TextAlign, style fyne.TextStyle) *Label {
```

Comment:
We have used the format NewEntry, NewPasswordEntry and NewMultiLineEntry elsewhere, I wonder if this should be NewStyledLabel?

Reply:
I think that works :)